client/state/jetpack-settings/modules/selectors.js

```diff
@@ -37,3 +37,15 @@ export function isActivatingModule( state, siteId, moduleSlug ) {
 export function isDeactivatingModule( state, siteId, moduleSlug ) {
 	return get( state.jetpackSettings.jetpackModules.requests, [ siteId, moduleSlug, 'deactivating' ], null );
 }
+
+/**
+ * Returns true if we are currently making a request to get the list of Jetpack
+ * modules on the site
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {String}  siteId The ID of the site we're querying
+ * @return {?Boolean}         Whether the list is being requested
+ */
+export function isFetchingModuleList( state, siteId ) {
+	return get( state.jetpackSettings.jetpackModules.requests, [ siteId, 'isFetchingModuleList' ], null );
```

Comment:
One thing we might want to think about here is consistency of the request property names. For the module activation/deactivation, they're called `activating` and `deactivating`, while here the prop is `isFetchingModuleList`. We might want to stay consistent, and either always prefix them with `is`, or never prefix them.